pkg/compute/publish.go

```diff
@@ -0,0 +1,98 @@
+package compute
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/text"
+)
+
+// PublishCommand produces and deploys an artifact from files on the local disk.
+type PublishCommand struct {
+	common.Base
+	manifest manifest.Data
+	build    *BuildCommand
+	deploy   *DeployCommand
+
+	// Deploy fields
+	path    common.OptionalString
+	version common.OptionalInt
+
+	// Build fields
+	name       common.OptionalString
+	lang       common.OptionalString
+	includeSrc common.OptionalBool
+	force      common.OptionalBool
+}
+
+// NewPublishCommand returns a usable command registered under the parent.
+func NewPublishCommand(parent common.Registerer, globals *config.Data, build *BuildCommand, deploy *DeployCommand) *PublishCommand {
+	var c PublishCommand
+	c.Globals = globals
+	c.manifest.File.SetOutput(c.Globals.Output)
+	c.manifest.File.Read(manifest.Filename)
+	c.build = build
+	c.deploy = deploy
+	c.CmdClause = parent.Command("publish", "Composite of \"build\", then \"deploy\"")
+
+	// Deploy flags
+	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
+	c.CmdClause.Flag("version", "Number of version to activate").Action(c.version.Set).IntVar(&c.version.Value)
+	c.CmdClause.Flag("path", "Path to package").Short('p').Action(c.path.Set).StringVar(&c.path.Value)
+
+	// Build flags
+	c.CmdClause.Flag("name", "Package name").Action(c.name.Set).StringVar(&c.name.Value)
+	c.CmdClause.Flag("language", "Language type").Action(c.lang.Set).StringVar(&c.lang.Value)
+	c.CmdClause.Flag("include-source", "Include source code in built package").Action(c.includeSrc.Set).BoolVar(&c.includeSrc.Value)
+	c.CmdClause.Flag("force", "Skip verification steps and force build").Action(c.force.Set).BoolVar(&c.force.Value)
```

Comment:
By redefining the flags for each sub-command we risk them quickly diverging if a new flag is added to the real command. Couple of thoughts:

- Could we access the sub-commands flags programmatically via the `CmdClause` structure?  My memory seems to think this isn't possible as the flaggroup and argGroup fields are private, but worth a proper investigation. 
- If not, can we add a comment in the parent commands to ensure future developers update this list
- If not, can we write some kind of test which will fail if the commands diverge?

Reply:
The ordering of these is also weird to me, as we build before we deploy.

Reply:
Good point about the divergence issue.

A quick print of the `BuildCommand.CmdClause` (for example) shows the fields are indeed private:

```go
&{
  cmdMixin:{
    actionMixin:{actions:[] preActions:[]} 
    flagGroup:0xc000275b00 
    argGroup:0xc000210d80 
    cmdGroup:0xc000275bc0
  } 
  app:0xc00026f9e0 
  name:build 
  aliases:[] 
  help:Build a Compute@Edge package locally 
  isDefault:false 
  validator:<nil> 
  hidden:false 
  completionAlts:[] 
  optionalSubcommands:false
}
```

So I think I'll investigate the failing test scenario first, and if that's not reasonable to achieve then I'll comment in the code on the coupling.

Reply:
Added a test.